src/libraries/System.Reflection.Metadata/src/System/Reflection/Metadata/BlobBuilder.cs

```diff
@@ -396,6 +411,7 @@ public void LinkPrefix(BlobBuilder prefix)
             // avoid chaining empty chunks:
             if (prefix.Count == 0)
             {
+                _emptyChildBlobs.Add(prefix);
```

Comment:
Why did you feel the need to create a collection here? My intuition is that this line just became `prefix.Free()`. There is no need to delay the `Free` at this point cause the `BlobBuilder` is considered unusable by the caller and the callee has decided it's going to be unused.

Reply:
From a user point of view, it would be surprising to Link a blob and found out that it was freed. That's why I preferred to delay freeing until calling Free method.

Reply:
What user is expecting that?